controllers/controllers.go

```diff
@@ -84,6 +119,51 @@ func (r *instanceReconciler) ensureInstanceIsUpToDate(instanceInfo *instance.Inf
 	return nc.Configure()
 }
 
+// upgradeBlocked returns whether the upgrade should be blocked if an upgrade will break storage functionality.
+func (r *instanceReconciler) upgradeBlocked(node *core.Node, upgradingToCSI bool) bool {
+	// Only consider the case of vSphere and Azure. Safely migrating in-tree storage on other platforms is not supported
+	// by WMCO.
+	if r.platform != config.VSpherePlatformType && r.platform != config.AzurePlatformType {
+		return false
+	}
+	if !upgradingToCSI {
+		return false
+	}
+	if _, present := node.GetLabels()[AllowUpgradeLabel]; present {
```

Comment:
The `blockMessage` above says that the label's value should be `=true` but here it seems the value isn't important, only the label's presence. Maybe update the block message to be consistent